sdk/sdk.go

```diff
@@ -112,12 +114,21 @@ func New(platformEndpoint string, opts ...Option) (*SDK, error) {
 		dialOptions = append(dialOptions, cfg.extraDialOptions...)
 	}
 
-	// IF IPC is disabled we build a connection to the platform
+	// IF IPC is disabled we build a validated healthy connection to the platform
 	if !cfg.ipc {
 		platformEndpoint, err = SanitizePlatformEndpoint(platformEndpoint)
 		if err != nil {
 			return nil, fmt.Errorf("%w [%v]: %w", ErrPlatformEndpointMalformed, platformEndpoint, err)
 		}
+		
+		// TODO: to support an offline SDK, we would need to remove the connection requirement
+		// For now we will only skip for test purposes
+		if !cfg.testSkipValidatePlatformConnectivity {
```

Comment:
Why not add a `WithConnectionValidation` option to the constructor? This would keep the current behavior of the SDK object (arguably this is a semver breaking change) and allow you to add the feature in otdfctl or other app integrators as needed

Reply:
I could also move this connectivity check inside the case where `WithPlatformConfiguration` was not utilized, but it feels like that option is actually driving a code path implementing something like `WithUnvalidatedPlatform`. 🤷